resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/security/HadoopFSCredentialProvider.scala

```diff
@@ -48,9 +50,16 @@ private[security] class HadoopFSCredentialProvider
     val tmpCreds = new Credentials()
     val tokenRenewer = getTokenRenewer(hadoopConf)
     hadoopFSsToAccess(hadoopConf, sparkConf).foreach { dst =>
-      val dstFs = dst.getFileSystem(hadoopConf)
-      logInfo("getting token for: " + dst)
-      dstFs.addDelegationTokens(tokenRenewer, tmpCreds)
+      try {
+        val dstFs = dst.getFileSystem(hadoopConf)
+        logInfo("getting token for: " + dst)
+        dstFs.addDelegationTokens(tokenRenewer, tmpCreds)
+      } catch {
+        case e: StandbyException =>
+          logWarning(s"Namenode ${dst} is in state standby", e)
```

Comment:
Hadoop compatible FS doesn't equal to HDFS, we can configure to wasb, adsl and others. Also wasb and adsl support fetching delegation tokens from common FS API, so we should avoid mentioning Namenode which is only existed in HDFS.